### PR TITLE
Use new api to display runtime stacks

### PR DESCRIPTION
**Pull request by EricJizbaMSFT**
The main point of this PR is to replace the hard-coded stacks in `functionStacks.ts` with a dynamic API call. The schema is a bit different, but the functionality is pretty much the exact same.

Fixes https://github.com/microsoft/vscode-azurefunctions/issues/2325
Fixes https://github.com/microsoft/vscode-azurefunctions/issues/2385
Fixes https://github.com/microsoft/vscode-azurefunctions/issues/2482
Fixes https://github.com/microsoft/vscode-azurefunctions/issues/2494

### CODE REVIEW

File: src/commands/createFunctionApp/stacks/FunctionAppStackStep.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+export class FunctionAppStackStep extends AzureWizardPromptStep<IFunctionAppWizardContext> {
+    public async prompt(context: IFunctionAppWizardContext): Promise<void> {
+        const placeHolder: string = localize('selectRuntimeStack', 'Select a runtime stack.');
+        context.newSiteStack = (await ext.ui.showQuickPick(getStackPicks(context), { placeHolder })).data;
+
+        if (!context.newSiteStack.minorVersion.stackSettings.linuxRuntimeSettings) {
+            context.newSiteOS = WebsiteOS.windows;
+        } else if (!context.newSiteStack.minorVersion.stackSettings.windowsRuntimeSettings) {
+            context.newSiteOS = WebsiteOS.linux;
+        } else if (!context.advancedCreation) {
```
Comment (nturinski) [inline src/commands/createFunctionApp/stacks/FunctionAppStackStep.ts:22]: Shouldn't you default to the `preferredOs` first and if it's advanced creation, then check the `minorVersion.stackSettings`?

File: src/commands/createFunctionApp/stacks/FunctionAppStackStep.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    public shouldPrompt(context: IFunctionAppWizardContext): boolean {
+        return !context.newSiteStack;
+    }
+
+    public async getSubWizard(context: IFunctionAppWizardContext): Promise<IWizardOptions<IFunctionAppWizardContext> | undefined> {
+        if (context.newSiteOS === undefined) {
+            return { promptSteps: [new SiteOSStep()] };
+        } else {
+            await setLocationsTask(context);
```
Comment (nturinski) [inline src/commands/createFunctionApp/stacks/FunctionAppStackStep.ts:35]: This seems like a behavioral change.  Before `setLocationsTask` was always called in SubscriptionTreeItem if  `!advancedCreation` but `newSiteOS` can be set in either scenario, right?

File: src/commands/createFunctionApp/stacks/FunctionAppStackStep.ts (hunk 1 above)
Comment (EricJizbaMSFT) [reply src/commands/createFunctionApp/stacks/FunctionAppStackStep.ts:22]: Realistically the order here won't make a difference. The point of the above checks is to see if that stack even supports both OS's - if it only supports one OS, we just use that OS. It should be impossible for the "preferredOS" to be an OS that the stack doesn't support

File: src/commands/createFunctionApp/stacks/FunctionAppStackStep.ts (hunk 2 above)
Comment (EricJizbaMSFT) [reply src/commands/createFunctionApp/stacks/FunctionAppStackStep.ts:35]: It's also set in `SiteOSStep` [here](https://github.com/microsoft/vscode-azuretools/blob/main/appservice/src/createAppService/SiteOSStep.ts#L21). In the past we _always_ prompted for OS during advanced create, but now we don't. Thus I have to `setLocationsTask` here to maintain existing behavior.

Comment (EricJizbaMSFT) [reply src/commands/createFunctionApp/stacks/FunctionAppStackStep.ts:35]: Err actually that was true for App Service - Functions is a bit different. It'll overwrite the locationsTask after the "HostPlan" step [here](https://github.com/microsoft/vscode-azurefunctions/blob/b6d5cd3a140338de5351fdeefafd33ef65af1213/src/commands/createFunctionApp/FunctionAppHostingPlanStep.ts#L21) (which happens after we prompt for runtime/OS).

So setting it here isn't technically necessary, but it's consistent with App Service and doesn't really hurt anything.

**Review by nturinski — APPROVED** [review]

**Pull request merged**